beacon-chain/rpc/validator_server.go

```diff
@@ -450,24 +455,26 @@ func (vs *ValidatorServer) addNonActivePublicKeysAssignmentStatus(
 	return assignments
 }
 
-func (vs *ValidatorServer) depositBlockSlot(ctx context.Context, eth1BlockNumBigInt *big.Int,
-	beaconState *pbp2p.BeaconState) (uint64, error) {
-	eth1BlockNum := eth1BlockNumBigInt.Uint64()
-	addFollowDistance := eth1BlockNum + params.BeaconConfig().Eth1FollowDistance
-	eth1Timestamp, err := vs.powChainService.BlockTimeByHeight(ctx, big.NewInt(int64(addFollowDistance)))
+func (vs *ValidatorServer) depositBlockSlot(ctx context.Context, currentSlot uint64,
+	eth1BlockNumBigInt *big.Int, beaconState *pbp2p.BeaconState) (uint64, error) {
+	blockTimeStamp, err := vs.powChainService.BlockTimeByHeight(ctx, eth1BlockNumBigInt)
 	if err != nil {
 		return 0, err
 	}
+	followTime := time.Duration(params.BeaconConfig().Eth1FollowDistance*params.BeaconConfig().GoerliBlockTime) * time.Second
+	eth1UnixTime := time.Unix(int64(blockTimeStamp), 0).Add(followTime)
 
 	votingPeriodSlots := helpers.StartSlot(params.BeaconConfig().EpochsPerEth1VotingPeriod)
 	votingPeriodSeconds := time.Duration(votingPeriodSlots*params.BeaconConfig().SecondsPerSlot) * time.Second
-
-	eth1UnixTime := time.Unix(int64(eth1Timestamp), 0)
 	timeToInclusion := eth1UnixTime.Add(votingPeriodSeconds)
 
 	eth2Genesis := time.Unix(int64(beaconState.GenesisTime), 0)
 	eth2TimeDifference := timeToInclusion.Sub(eth2Genesis).Seconds()
 	depositBlockSlot := uint64(eth2TimeDifference) / params.BeaconConfig().SecondsPerSlot
 
+	if depositBlockSlot > currentSlot-params.BeaconConfig().GenesisSlot {
+		return 0, errors.New("estimated Slot is in the future not the past")
```

Comment:
```suggestion
		return 0, nil
```

I dont think this should return an error. Is this error visible to the user?